### PR TITLE
test: migrate TcpProxy test configurations to v2

### PR DESCRIPTION
**Pull request by derekargueta**
Description: part of issue #6362
Risk Level: low
Testing: existing
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: Derek Argueta <dereka@pinterest.com>

### CODE REVIEW

**Comment by derekargueta** [comment]
@alyssawilk @zuercher (tcp_proxy owners) FYI the tests are failing because I'm using a deprecated field (`deprecated_v1`) to maintain test parity with the v1 configurations. It was deprecated in May with 1.11.0 so we still have a while before it can be removed. I'm assuming we don't want to just remove these tests as it would negatively impact test coverage. Can we just mark this PR as no-stale and when `deprecated_v1` is ready to be removed I can tack it on here with the removal of the tests?

**Comment by lizan** [comment]
@derekargueta wrap the test name with `DEPRECATED_FEATURE_TEST` macro then it will pass CI and tested only when deprecated feature is not disabled at compile time.

**Comment by derekargueta** [comment]
@lizan oh nice, thanks for the tip

File: test/common/tcp_proxy/tcp_proxy_test.cc [hunk 1]
```diff
@@ -48,109 +51,98 @@ namespace {
 using ::Envoy::Network::UpstreamServerName;
 
 namespace {
-Config constructConfigFromJson(const Json::Object& json,
+Config constructConfigFromYaml(const std::string& json,
```
Comment (zuercher) [inline test/common/tcp_proxy/tcp_proxy_test.cc]: Let's rename the parameter as well.

**Comment by mattklein123** [comment]
Epic!

**Pull request merged**